src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SerializableNamingRule.cs

```diff
@@ -5,16 +5,23 @@
 #nullable disable
 
 using System;
+using System.Runtime.Serialization;
 using System.Xml.Linq;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
 {
-    internal sealed class SerializableNamingRule : IEquatable<SerializableNamingRule>, IObjectWritable
+    [DataContract]
+    internal sealed record class SerializableNamingRule : IObjectWritable
     {
-        public Guid SymbolSpecificationID;
-        public Guid NamingStyleID;
-        public ReportDiagnostic EnforcementLevel;
+        [DataMember(Order = 0)]
+        public Guid SymbolSpecificationID { get; init; }
+
+        [DataMember(Order = 1)]
+        public Guid NamingStyleID { get; init; }
+
+        [DataMember(Order = 2)]
+        public ReportDiagnostic EnforcementLevel { get; init; }
```

Comment:
❔ Why not `{ get; }`?

Reply:
`SerializableNamingRule` does not have an explicit constructor. `init;` is to make this object immutable, is not related to message-pack. An alternative is to write an explicit constructor.